### ISSUE TITLE
React native accessibility - the screen reader should read the text first which is passed by AccessibilityInfo.announceForAccessibility().

### ISSUE THREAD

**Issue by suresh-kommuri** [facebook/react-native#33190]
### Description

Using react native swiper, I need to focus on the desired react element and it should read first by the screen reader when the page loads.

- I used AccessibilityInfo.announceForAccessibility() but by using this on any event the screen reader is reading the first or last element on the screen and then it is reading the AccessibilityInfo.announceForAccessibility string.
- I used AccessibilityInfo.setAccessibilityFocus(reactTag), but it is not consistent to focus the element.

Similar question found in - https://stackoverflow.com/questions/64580176/react-native-accessibility-set-focus

### Version

0.67.1

### Output of `npx react-native info`

System:
    OS: macOS 11.6.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 135.33 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.16 - ~/.nvm/versions/node/v14.19.0/bin/npm
    Watchman: 2022.02.07.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: ^6.3.1 => 6.4.0 
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.1 => 0.67.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Using react native swiper, I need to focus on the desired react element and it should read first by the screen reader when the page loads.

- I used AccessibilityInfo.announceForAccessibility() but by using this on any event the screen reader is reading the first or last element on the screen and then it is reading the AccessibilityInfo.announceForAccessibility string.

Similar question found in - https://stackoverflow.com/questions/64580176/react-native-accessibility-set-focus

**Comment by Jiandong-Guo** [comment]
Is the AccessibilityInfo.announceForAccessibility(str) working for iOS?

**Comment by blavalla** [comment]
I'm not familiar with react-native-swiper (this is a third-party component), but it sounds like this may be a problem specific to how they are handling the callbacks for when a page is done being swiped into view.

Generally speaking, you had the right approach by using `AccessibilityInfo.setAccessibilityFocus()`, as this method should move focus to a given element. If it's not working, it may actually be that focus is getting lost due to the element being updated in some way.

An easy thing to try to narrow down the problem would be to add a slight delay to your focus call to make sure that the swiper component is done updating before attempting to move focus to it. Something like 500ms should be enough. If that does fix the issue, then this is definitely a problem with react-native-swiper and when its informing you that the new view is ready.

I don't think there's anything to be done on the react native side though, so I'm going to close this out.  Feel free to re-open if you find that this is an issue with RN specifically, and not the react-native-swiper component.
